lucene/core/src/test/org/apache/lucene/search/TestFieldSortOptimizationSkipping.java

```diff
@@ -432,7 +439,48 @@ public void testDocSortOptimization() throws IOException {
       assertTrue(topDocs.totalHits.value < 10); // assert that very few docs were collected
     }
 
+    reader.close();
+    dir.close();
+  }
+
+  /**
+   * Test that sorting on _doc works correctly.
+   * This test goes through DefaultBulkSorter::scoreRange, where scorerIterator is BitSetIterator.
+   * As a conjunction of this BitSetIterator with DocComparator's iterator, we get BitSetConjunctionDISI.
+   * BitSetConjuctionDISI advances based on the DocComparator's iterator, and doesn't consider
+   * that its BitSetIterator may have advanced passed a certain doc. 
```

Comment:
Should we consider this a bug in `BitSetIterator` ?

Reply:
I will create an issue for this.

Reply:
Issue created: https://issues.apache.org/jira/browse/LUCENE-9541